compiler/riscv/codegen/ControlFlowEvaluator.cpp

```diff
@@ -502,6 +502,45 @@ OMR::RV::TreeEvaluator::acmpeqEvaluator(TR::Node *node, TR::CodeGenerator *cg)
 	return OMR::RV::TreeEvaluator::unImpOpEvaluator(node, cg);
 	}
 
+// also handles lselect, aselect, bselect, sselect
+TR::Register *
+OMR::RV::TreeEvaluator::iselectEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Node *condNode = node->getChild(0);
+   TR::Node *trueNode   = node->getChild(1);
+   TR::Node *falseNode  = node->getChild(2);
+
+   TR::Register *condReg = cg->evaluate(condNode);
+   TR::Register *trueReg = cg->gprClobberEvaluate(trueNode);
+   TR::Register *falseReg = cg->evaluate(falseNode);
+   TR::RealRegister *zero = cg->machine()->getRealRegister(TR::RealRegister::zero);
+
+
+   TR::LabelSymbol *startLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *joinLabel = generateLabelSymbol(cg);
+
+   startLabel->setStartInternalControlFlow();
+   joinLabel->setEndInternalControlFlow();
+
+   TR::RegisterDependencyConditions *deps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(3, 3,cg->trMemory());
```

Comment:
Minor point, but you only need to add post dependencies for ICF. The pre dependencies are not needed.

Reply:
Okay. 
I must admit I'm not very familiar with details of RA. This is essentially the same code as in 
`?min` / `?max` evaluators from PR #4894 
https://github.com/eclipse/omr/blob/455c7f2f1b12ab6308d151af0868696a5c41460a/compiler/riscv/codegen/ControlFlowEvaluator.cpp#L591-L593 
which follows POWER code as suggested by @aviansie-ben 
https://github.com/eclipse/omr/blob/cc446f6156db256c28d0cef1e4284bc730ec6677/compiler/p/codegen/FPTreeEvaluator.cpp#L138-L140

So, shall I fix it in `?min` / `?max` too?

Reply:
Yeah you can fix it there as well. I assume you copied what Power does right? I'm not sure why we add both post and pre dependencies on Power. @aviansie-ben do you might know? It is my understanding that for ICF regions, adding both to the merge label serves no purpose when only post conditions will do.

Reply:
As an example all ICF regions on Z use only post dependency conditions.

Reply:
Yeah, I shamelessly copied Power code as I did not / do not know what I'm doing :-)
OK, will fix it in both in this PR.

Reply:
The code is not adding only post conditions however we are still allocating preconditions (`3, 3`). Can we change that to `0, 3`?

Reply:
Sure! Sorry.

Reply:
No worries. Thanks!

Reply:
@fjeremic I've asked around and it looks like you're right. This is a historical oddity that should be corrected. I've opened #4991 to get this fixed in the Power codegen.